pkg/resource/edit/operations_test.go

```diff
@@ -158,6 +158,46 @@ func TestDeletingDuplicateURNs(t *testing.T) {
 	})
 }
 
+func TestDeletingDuplicateProviderURN(t *testing.T) {
+	t.Parallel()
+
+	// Create duplicate provider resources
+	pA := NewProviderResource("a", "p1", "0")
+	pB := NewProviderResource("a", "p2", "0")
+
+	// Create a resource that depends on the duplicate Provider.
```

Comment:
This isn't a duplicate? The names are different? Should they be the same but just different IDs?

Reply:
Yes I was being a dummy and misread the `NewProviderResource` function.  They should indeed be the same with just different IDs.